compiler/rustc_session/src/session.rs

```diff
@@ -1710,6 +1710,15 @@ impl EarlyErrorHandler {
         self.handler.struct_fatal(msg).emit()
     }
 
+    #[allow(rustc::untranslatable_diagnostic)]
+    #[allow(rustc::diagnostic_outside_of_impl)]
+    pub(crate) fn early_struct_error(
```

Comment:
It's unfortunate that I had to introduce this method. I can get rid of it again at the cost of not showing the `.help("replace dashes with underscores …")`.

Reply:
It's fine.